CMakeLists.txt

```diff
@@ -140,6 +140,14 @@ cmake_dependent_option(FUZZ "Build for fuzzing. Enabling this will disable all o
 
 option(INSTALL_MAN "Install man pages." ON)
 
+set(APPEND_CPPFLAGS "" CACHE STRING "Preprocessor flags that are appended to the flags added by the build system.")
+set(APPEND_CFLAGS "" CACHE STRING "C compiler flags that are appended to the flags added by the build system.")
+set(APPEND_CXXFLAGS "" CACHE STRING "(Objective) C++ compiler flags that are appended to the flags added by the build system.")
+set(APPEND_LDFLAGS "" CACHE STRING "Linker flags that are appended to the flags added by the build system.")
+string(APPEND CMAKE_CXX_COMPILE_OBJECT " ${APPEND_CPPFLAGS} ${APPEND_CXXFLAGS}")
+string(APPEND CMAKE_CXX_CREATE_SHARED_LIBRARY " ${APPEND_LDFLAGS}")
+string(APPEND CMAKE_CXX_LINK_EXECUTABLE " ${APPEND_LDFLAGS}")
```

Comment:
Why not append to both `CMAKE_OBJCXX_COMPILE_OBJECT` and `CMAKE_C_COMPILE_OBJECT` here as well? With the current approach if a new `.c` file is added in `src`, outside of `secp256k1`, then it will not get the append flags.

Reply:
The `CMAKE_<LANG>_COMPILE_OBJECT` variables are unset before enabling a language `<LANG>`, which means there is nothing to modify at this point.